### ISSUE TITLE
IP's allocated from original pool after deletion

### ISSUE THREAD

**Issue by caseydavenport** [projectcalico/libcalico-go#258]
This seems to be a reoccurrence of #56. I used https://github.com/projectcalico/calico/blob/master/v2.0/getting-started/kubernetes/installation/hosted/calico.yaml to deploy into a kops cluster.

System pods get 192.168.0.0/16 ip's:
```
admin@ip-172-20-83-74:~$ kubectl get po -n kube-system -o wide
NAME                                                                 READY     STATUS    RESTARTS   AGE       IP                NODE
...
dns-controller-2522975163-kbdx6                                      1/1       Running   0          2m        192.168.138.128   ip-172-20-83-74.us-west-1.compute.internal
kube-dns-v20-3531996453-16zaj                                        2/3       Running   0          1h        192.168.140.0     ip-172-20-89-244.us-west-1.compute.internal
kube-dns-v20-3531996453-l9px7                                        3/3       Running   0          1h        192.168.149.128   ip-172-20-117-143.us-west-1.compute.internal
...
```

Default pool is deleted and new pool is created:
```
admin@ip-172-20-83-74:~$ calicoctl delete IpPool 192.168.0.0/16
Successfully deleted 1 'ipPool' resource(s)
admin@ip-172-20-83-74:~$ cat << EOF | calicoctl create -f -
 - apiVersion: v1
>   kind: ipPool
>   metadata:
>     cidr: 100.64.0.0/10
>   spec:
>     ipip:
>       enabled: true
>     nat-outgoing: true
> EOF
Successfully created 1 'ipPool' resource(s)
```

System pods deleted:
```
admin@ip-172-20-83-74:~$ kubectl delete po dns-controller-2522975163-kbdx6 kube-dns-v20-3531996453-16zaj kube-dns-v20-3531996453-l9px7 -n kube-system
pod "dns-controller-2522975163-kbdx6" deleted
pod "kube-dns-v20-3531996453-16zaj" deleted
pod "kube-dns-v20-3531996453-l9px7" deleted
```

New system pods still have ip's from 192.168.0.0/16:
```
admin@ip-172-20-83-74:~$ kubectl get po -n kube-system -o wide
NAME                                                                 READY     STATUS        RESTARTS   AGE       IP                NODE
...
dns-controller-2522975163-dczfr                                      1/1       Running       0          3s        192.168.138.129   ip-172-20-83-74.us-west-1.compute.internal
kube-dns-v20-3531996453-16zaj                                        3/3       Terminating   0          1h        192.168.140.0     ip-172-20-89-244.us-west-1.compute.internal
kube-dns-v20-3531996453-c7z23                                        2/3       Running       0          3s        192.168.140.1     ip-172-20-89-244.us-west-1.compute.internal
kube-dns-v20-3531996453-l9px7                                        3/3       Terminating   0          1h        192.168.149.128   ip-172-20-117-143.us-west-1.compute.internal
kube-dns-v20-3531996453-ujdum                                        2/3       Running       0          3s        192.168.149.129   ip-172-20-117-143.us-west-1.compute.internal
...
```

**Comment by caseydavenport** [comment]
Moving this from https://github.com/projectcalico/libcalico/issues/169

CC @stonith

**Comment by gunjan5** [comment]
Fixed in #264
